src/drivers/barometer/bmp280/bmp280_main.cpp

```diff
@@ -173,6 +191,8 @@ static int usage()
 	PX4_INFO("    -I    (i2c internal bus)");
 	PX4_INFO("    -s    (spi internal bus)");
 	PX4_INFO("    -S    (spi external bus)");
+	PX4_INFO("    -m    spi_bus_mode (0-3)");
```

Comment:
Why is setting the spi mode on the command line necessary?

Reply:
The host system currently cannot support default SPI Mode 3, so a way to use other mode is needed

Reply:
Why can't the host system change the SPI mode?

In the case of the bmp280 it appears to support both SPI mode 0 and 3. We could also simply change the only supported bmp280 spi mode depending on where this requirement is coming from.

Reply:
SPI mode 3 currently does not work on the host system. With mode 3, it cannot go pass probe (whoami) phase. But the host system is currently working with spi mode 0. Changing default spi mode of bmp280 to spi mode 0 might break other systems.

Reply:
Correct SPI communication between SPI master and slaves highly depends on maximum supported frequency and relative timing of CLK, CS, MISO and MOSI signal, so no matter what kind of host system is in use, it's better to have option to customize bus frequency and SPI mode. I guess the intent to set SPI mode in board configuration is to apply the same SPI mode to all SPI sensors attached to the board, but it cannot deal with the case that some sensors work with only one SPI mode while others only work with a different SPI mode. Setting SPI mode at individual sensor level is more flexible as it allows choosing the best matching mode between master and slave for an individual sensor.

Reply:
> I guess the intent to set SPI mode in board configuration is to apply the same SPI mode to all SPI sensors attached to the board, but it cannot deal with the case that some sensors work with only one SPI mode while others only work with a different SPI mode.

Not necessarily, we can also do it per-device. I'm working on some general improvements around that in https://github.com/PX4/Firmware/pull/14156. I don't have a clear preference.